napari/utils/color.py

```diff
@@ -15,6 +15,11 @@ class ColorValue(np.ndarray):
     use the ``validate`` method to coerce a value to a single color.
     """
 
+    def __new__(cls, input_array):
+        # see https://numpy.org/doc/stable/user/basics.subclassing.html#slightly-more-realistic-example-attribute-added-to-existing-array
+        obj = np.asarray(input_array).view(cls)
```

Comment:
I considered this in [PR 4704](https://github.com/napari/napari/pull/4704#discussion_r900289069), though was considering something slightly fancier. Agreed that we should either do something like this or remove the `np.ndarray` as a sub-type (to indicate that this is just a pydantic field with some specific coercion logic) and I prefer doing this (probably because it plays nice with type checking in pydantic and in general).

One suggestion is actually to call `validate` instead of `asarray` here and in the other similar classes.

```suggestion
        obj = cls.validate(input_array).view(cls)
```

That's because I'd expect something like `ColorValue([0, 1])` to fail and maybe something like `ColorValue('red')` to do the coercion.

I guess this PR is an improvement as we can actually use the constructor more easily now, but doesn't quite feel complete.

Reply:
Or will that cause `validate` to be called twice now? If so should we consider removing `__get_validators__`?

Reply:
Actually no, it wouldn't be called twice because of this check in `coerce_type`:

```py
if isinstance(value, origin):
	return value
```